repos/system_upgrade/el9toel10/actors/checkoldxfs/libraries/checkoldxfs.py

```diff
@@ -0,0 +1,141 @@
+from leapp import reporting
+from leapp.exceptions import StopActorExecutionError
+from leapp.libraries.stdlib import api
+from leapp.models import XFSInfoFacts
+
+RHEL_9_TO_10_BACKUP_RESTORE_LINK = 'https://red.ht/rhel_9_to_10_backup_restore_xfs'
+
+FMT_LIST_SEPARATOR = '\n    - '
+
+
+def _formatted_list_output(input_list, sep=FMT_LIST_SEPARATOR):
+    return ['{}{}'.format(sep, item) for item in input_list]
+
+
+def process():
+    xfs_info_facts = _get_xfs_info_facts()
+
+    invalid_bigtime = []
+    invalid_crc = []
+    for xfs_info in xfs_info_facts.mountpoints:
+        if not _has_valid_bigtime(xfs_info):
+            api.current_logger().debug(
+                'Mountpoint {} has invalid bigtime'.format(xfs_info.mountpoint)
+            )
+            invalid_bigtime.append(xfs_info.mountpoint)
+
+        if not _has_valid_crc(xfs_info):
+            api.current_logger().debug(
+                'Mountpoint {} has invalid crc'.format(xfs_info.mountpoint)
+            )
+            invalid_crc.append(xfs_info.mountpoint)
+
+    if invalid_bigtime or invalid_crc:
+        _inhibit_upgrade(invalid_bigtime, invalid_crc)
+
+        return
+
+    api.current_logger().debug('All XFS system detected are valid.')
+
+
+def _get_xfs_info_facts():
+    msgs = api.consume(XFSInfoFacts)
+
+    xfs_info_facts = next(msgs, None)
+    if xfs_info_facts is None:
+        raise StopActorExecutionError('Could not retrieve XFSInfoFacts!')
+
+    if next(msgs, None):
+        api.current_logger().warning(
+            'Unexpectedly received more than one XFSInfoFacts message.')
+
+    return xfs_info_facts
+
+
+def _has_valid_bigtime(xfs_info):
+    return xfs_info.meta_data.bigtime == '1'
+
+
+def _has_valid_crc(xfs_info):
+    return xfs_info.meta_data.crc == '1'
+
+
+def _inhibit_upgrade(invalid_bigtime, invalid_crc):
+    if invalid_bigtime:
+        _report_bigtime(invalid_bigtime)
+
+    if invalid_crc:
+        _inhibit_crc(invalid_crc)
+
+
+def _report_bigtime(invalid_bigtime):
+    title = 'Detected XFS filesystems without bigtime feature.'
+    summary = [(
+        'The XFS v5 filesystem format introduced the "bigtime" feature in RHEL 9,'
+        ' to support timestamps beyond the year 2038. XFS filesystems that'
+        ' do not have the "bigtime" feature enabled remain vulnerable to timestamp'
+        ' overflow issues. It is strongly recommended to enable this feature on all'
+        ' XFS filesystems to ensure long-term compatibility and prevent potential'
+        ' failures.'
+        ' Following XFS file systems have not enabled the "bigtime" feature:{}'
+        .format(''.join(_formatted_list_output(invalid_bigtime)))
+    )]
```

Comment:
```suggestion
    summary = (
        'The XFS v5 filesystem format introduced the "bigtime" feature in RHEL 9,'
        ' to support timestamps beyond the year 2038. XFS filesystems that'
        ' do not have the "bigtime" feature enabled remain vulnerable to timestamp'
        ' overflow issues. It is strongly recommended to enable this feature on all'
        ' XFS filesystems to ensure long-term compatibility and prevent potential'
        ' failures.'
        ' Following XFS file systems have not enabled the "bigtime" feature:{}'
        .format(''.join(_formatted_list_output(invalid_bigtime)))
    )
```

seeing it now, we should simplify it when the summary is not crafted from multiple blocks. similar for remediation hint, and the other report